scipy/optimize/tests/test_zeros.py

```diff
@@ -269,6 +269,69 @@ def test_root_scalar_fail(self):
         with pytest.raises(ValueError, match=message):
             root_scalar(f1, method='halley', fprime2=f1_2, x0=3, xtol=1e-6)  # no fprime
 
+    def test_array_chandrupatla(self):
```

Comment:
I will probably replace these, but in your next commit, would you move them into a separate class `TestChandrupatla` that comes before `TestNewton`?

Also, make sure they're working. Look at how `tstutils_functions` is defined at the top. I think `tstutils_functions[1]` is the same as the `f3` you were using before.